### PR TITLE
Closes #6685 - explain limitations of fixed arrays

### LINKED ISSUE #6685: More detail on removing items from Arrays
I was looking for details about removing items from Arrays, but didn't see it here.  On the PowerShell arrays there is a method "RemoveAt) that I attempted to use, but it returns an error indicating that arrays can't be modified.  Is this method supposed to be available?  If so, how should it be used?  Also, what about the Remove method?

Thanks!
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e99e524d-54e5-4119-9ebf-4fdbb7626122
* Version Independent ID: 271e2b41-af42-d4b5-3fdb-139aa5114b11
* Content: [Everything you wanted to know about arrays - PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-arrays?view=powershell-7)
* Content Source: [reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md)
* Product: **powershell**
* Technology: **powershell-conceptual**
* GitHub Login: @joeyaiello
* Microsoft Alias: **jaiello**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1777349](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1777349) - Closes #6685 - explain limitations of fixed arrays

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [x] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [ ] Version 7.1 preview content
- [ ] Version 7.0 content
- [ ] Version 6 content
- [ ] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md [hunk 1]
```diff
@@ -839,7 +839,11 @@ At this point, you're starting to wonder how to add items to an array. The quick
 can't. An array is a fixed size in memory. If you need to grow it or add a single item to it, then
 you need to create a new array and copy all the values over from the old array. This sounds
 expensive and like a lot of work, however, PowerShell hides the complexity of creating the new
```
Comment (chasewilson) [inline reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md:841]: What about something alonge the lines of:
```
This sounds expensive and like a lot of work. However, PowerShell hides the complexity of creating the new	array by implementing the addition operator (`+`). PowerShell does not implement a
subtraction operation.
```

Just seems to seem simpler but I don't know how you feel about changing the source material that much

Comment (chasewilson) [reply reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md:841]: Also, you can tell my english is on point by my use of "seems to seem..."

Comment (sdwheeler) [reply reference/docs-conceptual/learn/deep-dives/everything-about-arrays.md]: Made a minor tweak.

**Review by chasewilson — APPROVED** [review]

**Comment by chasewilson** [comment]
LGTM

**Pull request merged**